### PR TITLE
Queries: merge query_extensions into base_language_map

### PR DESCRIPTION
**Pull request by stsewd**
These two are doing the same currently.

### CODE REVIEW

File: lua/nvim-treesitter/query.lua [hunk 1]
```diff
@@ -13,12 +13,8 @@ local query_cache = caching.create_buffer_cache()
 M.base_language_map = {
   cpp = {'c'},
   typescript = {'javascript'},
-  tsx = {'typescript', 'javascript'},
-}
-
-M.query_extensions = {
-  javascript = { 'jsx' },
-  tsx = {'jsx'}
+  jsx = {'javascript'},
```
Comment (kyazdani42) [inline lua/nvim-treesitter/query.lua:16]: i believe this should be `javascript = { 'jsx' }`

**Review by kyazdani42 — CHANGES_REQUESTED** [review]

File: lua/nvim-treesitter/query.lua (hunk 1 above)
Comment (stsewd) [reply lua/nvim-treesitter/query.lua:16]: Should we do the same for typescript? Or do we need both relationshipts? `tsx = {'typescript'}` and `typescript = {'tsx'}`

Comment (kyazdani42) [reply lua/nvim-treesitter/query.lua:16]: tsx will be loaded when encountering typescriptreact, juste like javascriptreact will load javascript which will load the jsx query. Typescript parser does understand jsx queries so we should not do typescript = { 'tsx' }

Comment (stsewd) [reply lua/nvim-treesitter/query.lua]: Changed!

**Review by kyazdani42 — APPROVED** [review]
lgtm, let's merge

**Pull request merged**